src/hooks/useMessagesLane.ts

```diff
@@ -33,77 +33,84 @@ interface Props {
 }
 
 interface Output {
-  inboundLanes: {
-    bridgeReceivedMessages: string;
-  };
+  bridgeReceivedMessages: string;
   outboundLanes: {
     pendingMessages: string;
     totalMessages: string;
     latestReceivedNonce: string;
   };
 }
 
-const useMessageLane = ({ isApiReady, api, chain }: Props): Output => {
+const useMessagesLane = ({ isApiReady, api, chain }: Props): Output => {
+  const isMounted = useIsMounted();
   const { dispatchMessage } = useUpdateMessageContext();
   const [outboundLanes, setOutboudLanes] = useMountedState({
     latestReceivedNonce: '0',
     pendingMessages: '0',
     totalMessages: '0'
   });
-  const [inboundLanes, setInboudLanes] = useMountedState({ bridgeReceivedMessages: '0' });
+
+  const [bridgeReceivedMessages, setBridgesReceivedMessages] = useMountedState('0');
 
   const laneId = useLaneId();
-  const { bridgedMessages, latestReceivedNonceMethodName } = getSubstrateDynamicNames(chain);
 
-  useEffect((): (() => void) => {
-    const getLane = async (setter: any, isOutbound?: boolean) => {
-      try {
-        if (isOutbound) {
-          const u = await api.query[bridgedMessages].outboundLanes(laneId, (res: any) => {
-            const latest_generated_nonce = res.get('latest_generated_nonce').toString();
-            const latest_received_nonce = res.get('latest_received_nonce').toString();
-            const pendingMessages = new BN(latest_generated_nonce).sub(new BN(latest_received_nonce));
+  const { bridgedMessages } = getSubstrateDynamicNames(chain);
 
-            setter({
-              latestReceivedNonce: latest_received_nonce.toString(),
-              pendingMessages: pendingMessages.isNeg() ? '0' : pendingMessages.toString(),
-              totalMessages: latest_generated_nonce
-            });
+  useEffect(() => {
+    let unsubscribeOutboundLanes: () => void;
+    let unsubscribeInboundLanes: () => void;
+
+    if (api && isApiReady && api.query[bridgedMessages] && chain && isMounted()) {
+      api.query[bridgedMessages]
+        .outboundLanes(laneId, (res: any) => {
+          const latest_generated_nonce = res.get('latest_generated_nonce').toString();
+          const latest_received_nonce = res.get('latest_received_nonce').toString();
+          const pendingMessages = new BN(latest_generated_nonce).sub(new BN(latest_received_nonce));
+
+          setOutboudLanes({
+            latestReceivedNonce: latest_received_nonce.toString(),
+            pendingMessages: pendingMessages.isNeg() ? '0' : pendingMessages.toString(),
+            totalMessages: latest_generated_nonce
           });
-          return Promise.resolve(u);
-        }
-        const u = await api.query[bridgedMessages].inboundLanes(laneId, (res: any) => {
-          setter({ bridgeReceivedMessages: res.get('last_confirmed_nonce').toString() });
+        })
+        .then((unsub) => {
+          unsubscribeOutboundLanes = unsub;
```

Comment:
This is not preventing the leak issue that if the promise is not resolved by the time component is being unmounted, the unsub function will not run (anyway returning an async cleanup is not the right way either). Doing some research i found that the correct way to prevent this 100% is by using [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController).  Here it is a good explanation example about the usage: [link](https://dev.to/bil/using-abortcontroller-with-react-hooks-and-typescript-to-cancel-window-fetch-requests-1md4)

The limitation i see is that to be able to use the `abortController.abort()`, polkadotjs api need to support this by accepting a sign parameter so i can control the call with `abortController.signal`. Reading the api source code,  the api does not support this.

 Also the `useMountedState` prevents the issue of not updating the state if the component is not mounted but it is not preventing to execute the call if the component is already unmounted. So i have added back the `isMounted()` to prevent this. Actually polkadotjs UI is doing something similar: [useInactives](https://github.com/polkadot-js/apps/blob/master/packages/page-staking/src/Actions/useInactives.ts#L91)

@tomusdrw @wirednkod  i would like to have your feedback about this, sorry that i did not catch this during previous PR , but i realized about this when i was writing the tests.

Reply:
>  I though that there should be no API calls inside hooks - but all to be on a root level and been controlled by external functions. Did that change?

That's the idea, but that's a bigger refactor that i didn't want to include in this PR since it was going to make it even more bigger ( it's quite big )

Reply:
Also , i have opened this issue in polkadotjs api gh: https://github.com/polkadot-js/api/issues/3597